src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java

```diff
@@ -313,6 +317,22 @@ public Object convert(Class type, Object value) {
 
     }
 
+    /** A converter that converts a string array to a pattern. */
+    private static final class PatternArrayConverter implements Converter {
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public Object convert(Class type, Object value) {
+            final String plainString = value.toString();
+            final String[] patternStrings = plainString.split(REGEX_SEPARATOR);
```

Comment:
bad side effect of this is we should clearly mention that we do not allow `,` as part of regexp to match what user needs. So user can not use `,` in it Patterns, it is delimiter.
`\,` will also not work, as we simply splitting for just `,`.

Reply:
@romani, do you want me to close this PR for now? Since it might be confusing for others. We can re-open it later to further modify `Pattern[]` if it is the right time. :)

Reply:
Let's keep it open until we come to agreement on how to fix issue.